src/components/course/EnrollButton.jsx

```diff
@@ -63,6 +66,8 @@ export default function EnrollButton() {
     seats,
   } = activeCourseRun;
 
+  const [userSubsidy] = useFetchUserSubsidyForCourse(activeCourseRun, enterpriseConfig);
```

Comment:
This hook, which is responsible for making an API call, is also used in `<CourseSidebarPrice>`. If we use the hook in both `EnrollButton` and `CourseSidebarPrice`, we're making this API call twice on the same page.

I'd recommend moving the use of `useFetchUserSubsidyForCourse` up in the component tree such that only a single request for this data is made for the course page. The resulting data could be added to the `CourseContext` so it can be pulled from sub-components like `EnrollButton` and `CourseSidebarPrice`.

Reply:
We would be making the same call twice, but we have the caching working now (or so I thought). I was under the impression that we had decided to lean into that in order to avoid pulling up shared hooks

Reply:
I think we still want to be pulling up shared hooks in terms of cleaner code. The front end caching is more so about reducing repeated network calls between page loads as opposed to reducing network calls within the same page.

Reply:
Seems like a strange way to restrict caching for just reusing an existing hook, but okay. I will look into moving it

Reply:
Just picking back up on this, but so far my investigation makes me think that this way is going to be cleaner (not to mention lower effort) than the refactored version. I'm going to continue looking at it a little bit more, but I do want to make sure that we feel strongly about this before going too far down this path

Reply:
I'm fine with going with the cleanest version

Reply:
I do agree with Adam that the hook should be moved up in the component tree and shared between subcomponents using context

Reply:
> "To add additional detail, this felt like the cleaner approach than to avoid refactoring the initial state for the Course component as you can't use hooks in useMemo "

@bbaker6225 Do you mind clarifying this comment a bit?

My 2 cents about the current approach is that introducing an API call into the `EnrollButton` component is now adding async behavior which means that component can't be rendered until that data is loaded. This is why we have to show a skeleton loading state for the price details where we call the hook in `CourseSidebarPrice`. By using the hook in the `EnrollButton` (assuming the response hasn't been cached yet) for a user on a slow network, the `EnrollButton` will likely appear to "pop in" since we can't render it until that API call resolves.

By consolidating the 2 uses of the hook into one, and share the resulting data via React's Context, when we render the the course, we have all the data we need and don't need to worry about async loading of components popping into the UI.

Also, relying on the API cache alone here, is there a possibility that the two calls happen quick enough in succession where the data from the first call is not yet in the client cache and we still end up making 2 calls anyways?

Reply:
My comment was that we can't put the `userSubsidy` in the initial state for the course context since you can't use the hook in the `useMemo` call there. I can trying moving it into the course context provider, but that seemed like it would result in that same "popping in" behavior since it needs the course data to finish loading first. I also thought that it would be preferable from a UX perspective to have the initial page load be faster and have other parts come in later so the user can at least start to see the page (especially for the enroll button since I don't think people immediately hit that without learning anything else about the course). I could be wrong about that though, I'm no ux expert

For relying on the cache, I could see that as a possibility even though I haven't been able to reproduce it

Reply:
Adam and I talked about this more over slack and have a different direction to try that should satisfy both of these concerns

Reply:
Updated to use a different approach